modules/generator/storage/instance.go

```diff
@@ -98,6 +119,44 @@ func (s *storageImpl) Close() error {
 	).Err()
 }
 
+func (s *storageImpl) watchOverrides() {
+	t := time.NewTicker(30 * time.Second)
+	defer t.Stop()
+
+	for {
+		select {
+		case <-t.C:
+			newHeaders := s.overrides.MetricsGeneratorRemoteWriteHeaders(s.tenantID)
+			if !headersEqual(s.currentHeaders, newHeaders) {
+				level.Info(s.logger).Log("msg", "updating remote write headers")
+				s.currentHeaders = newHeaders
+				err := s.remote.ApplyConfig(&prometheus_config.Config{
+					RemoteWriteConfigs: generateTenantRemoteWriteConfigs(s.cfg.RemoteWrite, s.tenantID, newHeaders, s.cfg.RemoteWriteAddOrgIDHeader, s.logger),
+				})
+				if err != nil {
+					level.Error(s.logger).Log("msg", "Failed to update remote write headers. Remote write will continue with old headers", "err", err)
```

Comment:
do we know this is true? if it the prom remote storage fails to apply new configs do we know it continues with the old?

updateProcessors at the level above increments a counter metric on failure which i like. `metrics_generator_active_processors_update_failed_total`. i would love to just have all of this reloaded and counted under the same metric, but that name is pretty opinionated.

Reply:
>do we know this is true? if it the prom remote storage fails to apply new configs do we know it continues with the old?

Yes. If the update fails, the function returns and the original queue is left untouched https://github.com/prometheus/prometheus/blob/main/storage/remote/write.go#L148-L177.

>updateProcessors at the level above increments a counter metric on failure which i like. metrics_generator_active_processors_update_failed_total. i would love to just have all of this reloaded and counted under the same metric, but that name is pretty opinionated.

Good point. Added!